src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1686,77 +1672,60 @@ private void MakeSeparatorList(ReadOnlySpan<char> separators, ref ValueListBuild
 
         private void MakeSeparatorListVectorized(ref ValueListBuilder<int> sepListBuilder, char c, char c2, char c3)
         {
-            // Redundant test so we won't prejit remainder of this method
-            // on platforms without SSE.
-            if (!Sse41.IsSupported)
-            {
-                throw new PlatformNotSupportedException();
-            }
-
-            // Constant that allows for the truncation of 16-bit (FFFF/0000) values within a register to 4-bit (F/0)
-            Vector128<byte> shuffleConstant = Vector128.Create(0x00, 0x02, 0x04, 0x06, 0x08, 0x0A, 0x0C, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF);
-
-            Vector128<ushort> v1 = Vector128.Create((ushort)c);
-            Vector128<ushort> v2 = Vector128.Create((ushort)c2);
-            Vector128<ushort> v3 = Vector128.Create((ushort)c3);
-
-            ref char c0 = ref MemoryMarshal.GetReference(this.AsSpan());
-            int cond = Length & -Vector128<ushort>.Count;
             int i = 0;
-
-            for (; i < cond; i += Vector128<ushort>.Count)
+            if (Vector256.IsHardwareAccelerated && Length >= Vector256<ushort>.Count * 2)
             {
-                Vector128<ushort> charVector = ReadVector(ref c0, i);
-                Vector128<ushort> cmp = Sse2.CompareEqual(charVector, v1);
-
-                cmp = Sse2.Or(Sse2.CompareEqual(charVector, v2), cmp);
-                cmp = Sse2.Or(Sse2.CompareEqual(charVector, v3), cmp);
-
-                if (Sse41.TestZ(cmp, cmp)) { continue; }
-
-                Vector128<byte> mask = Sse2.ShiftRightLogical(cmp.AsUInt64(), 4).AsByte();
-                mask = Ssse3.Shuffle(mask, shuffleConstant);
+                ref ushort source = ref Unsafe.As<char, ushort>(ref _firstChar);
 
-                uint lowBits = Sse2.ConvertToUInt32(mask.AsUInt32());
-                mask = Sse2.ShiftRightLogical(mask.AsUInt64(), 32).AsByte();
-                uint highBits = Sse2.ConvertToUInt32(mask.AsUInt32());
+                Vector256<ushort> v1 = Vector256.Create((ushort)c);
+                Vector256<ushort> v2 = Vector256.Create((ushort)c2);
+                Vector256<ushort> v3 = Vector256.Create((ushort)c3);
 
-                for (int idx = i; lowBits != 0; idx++)
+                int vector256ShortCount = Vector256<ushort>.Count;
+                for (; (i + vector256ShortCount) <= Length; i += vector256ShortCount)
                 {
-                    if ((lowBits & 0xF) != 0)
+                    Vector256<ushort> vector = Vector256.LoadUnsafe(ref source, (uint)i);
+                    Vector256<ushort> cmp = Vector256.Equals(vector, v1) | Vector256.Equals(vector, v2) | Vector256.Equals(vector, v3);
+
+                    uint mask = cmp.AsByte().ExtractMostSignificantBits() & 0b0101010101010101;
```

Comment:
It might be a good idea to also use TestZ for faster out, e.g.
```csharp
if (cmp == Vector256<ushort>.Zero)
    continue;
```
it's faster than movmsk